backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -159,10 +159,20 @@ private Channel tryBorrowFromPool() {
         return Mono.fromCallable(this::borrowFromPool)
             .doOnError(throwable -> LOGGER.warn("Cannot borrow channel", throwable))
             .retryBackoff(MAX_BORROW_RETRIES, MIN_BORROW_DELAY, FOREVER, Schedulers.elastic())
+            .onErrorMap(this::propagateException)
             .subscribeOn(Schedulers.elastic())
             .block();
     }
 
+    private Throwable propagateException(Throwable throwable) {
+        if (throwable instanceof IllegalStateException
+            && throwable.getMessage().contains("Retries exhausted")) {
+            return throwable.getCause();
```

Comment:
why are you are throwing the cause?

Reply:
`Throwable` is being the `IllegalStateException: Retries exhausted 3/3` and it carries an actual exception. I do what It did before: throwing the actual exception, and brings more valuable information rather than the IllegalStateException thrown by `retryBackOff()`